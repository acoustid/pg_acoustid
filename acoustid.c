/* acoustid_compare.c */

#include "postgres.h"
#include "fmgr.h"

PG_MODULE_MAGIC;

PG_FUNCTION_INFO_V1(acoustid_compare);
Datum acoustid_compare(PG_FUNCTION_ARGS);

PG_FUNCTION_INFO_V1(acoustid_compare2);
Datum acoustid_compare2(PG_FUNCTION_ARGS);

PG_FUNCTION_INFO_V1(acoustid_compare3);
Datum acoustid_compare3(PG_FUNCTION_ARGS);

PG_FUNCTION_INFO_V1(acoustid_extract_query);
Datum acoustid_extract_query(PG_FUNCTION_ARGS);
