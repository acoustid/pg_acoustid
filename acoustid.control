# acoustid extension
comment = 'AcoustID utility functions'
default_version = '2.0'
module_pathname = '$libdir/acoustid'
relocatable = true
