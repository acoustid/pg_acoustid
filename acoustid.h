#ifndef PG_ACOUSTID_H_
#define PG_ACOUSTID_H_

#endif  // PG_ACOUSTID_H_
