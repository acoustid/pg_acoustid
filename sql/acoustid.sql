SET client_min_messages = warning;
\set ECHO none
\i acoustid.sql
\set ECHO all
RESET client_min_messages;

SELECT acoustid_compare('{}'::int4[], '{}'::int4[]);
SELECT acoustid_compare('{1}'::int4[], '{}'::int4[]);
SELECT acoustid_compare('{1}'::int4[], '{1}'::int4[]);
SELECT acoustid_compare('{1}'::int4[], '{2}'::int4[]);
SELECT acoustid_compare('{1}'::int4[], '{6}'::int4[]);
SELECT acoustid_compare('{1}'::int4[], '{1,6}'::int4[]);
SELECT acoustid_compare('{1,56}'::int4[], '{1,6}'::int4[]);
SELECT acoustid_compare('{448,56}'::int4[], '{1,6}'::int4[]);

SELECT acoustid_compare('{0,7,56,3584}'::int4[], '{0,7,56,448}'::int4[]);
SELECT acoustid_compare('{7,0,7,56,3584}'::int4[], '{0,7,56,448}'::int4[]);
SELECT acoustid_compare('{0,7,56,3584}'::int4[], '{7,0,7,56,448}'::int4[]);

SELECT acoustid_compare('{65535,65535,1,6,65535,65535}'::int4[], '{1,6}'::int4[]);
SELECT acoustid_compare('{1,6}'::int4[], '{65535,65535,1,6,65535,65535}'::int4[]);

SELECT acoustid_compare('{1,1,1,1,1}'::int4[], '{1,1,1,1,1}'::int4[]);

