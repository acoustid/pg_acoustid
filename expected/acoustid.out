SET client_min_messages = warning;
\set ECHO none
RESET client_min_messages;
SELECT acoustid_compare('{}'::int4[], '{}'::int4[]);
 acoustid_compare 
------------------
                0
(1 row)

SELECT acoustid_compare('{1}'::int4[], '{}'::int4[]);
 acoustid_compare 
------------------
                0
(1 row)

SELECT acoustid_compare('{1}'::int4[], '{1}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

SELECT acoustid_compare('{1}'::int4[], '{2}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

SELECT acoustid_compare('{1}'::int4[], '{6}'::int4[]);
 acoustid_compare 
------------------
                0
(1 row)

SELECT acoustid_compare('{1}'::int4[], '{1,6}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

SELECT acoustid_compare('{1,56}'::int4[], '{1,6}'::int4[]);
 acoustid_compare 
------------------
              0.5
(1 row)

SELECT acoustid_compare('{448,56}'::int4[], '{1,6}'::int4[]);
 acoustid_compare 
------------------
                0
(1 row)

SELECT acoustid_compare('{0,7,56,3584}'::int4[], '{0,7,56,448}'::int4[]);
 acoustid_compare 
------------------
             0.75
(1 row)

SELECT acoustid_compare('{7,0,7,56,3584}'::int4[], '{0,7,56,448}'::int4[]);
 acoustid_compare 
------------------
             0.75
(1 row)

SELECT acoustid_compare('{0,7,56,3584}'::int4[], '{7,0,7,56,448}'::int4[]);
 acoustid_compare 
------------------
             0.75
(1 row)

SELECT acoustid_compare('{65535,65535,1,6,65535,65535}'::int4[], '{1,6}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

SELECT acoustid_compare('{1,6}'::int4[], '{65535,65535,1,6,65535,65535}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

SELECT acoustid_compare('{1,1,1,1,1}'::int4[], '{1,1,1,1,1}'::int4[]);
 acoustid_compare 
------------------
                1
(1 row)

